vortex-io/src/object_store.rs

```diff
@@ -76,10 +77,59 @@ impl VortexReadAt for ObjectStoreReadAt {
 
         Box::pin(async move {
             let start_range = pos as usize;
-            let bytes = object_store
-                .get_range(&location, start_range..(start_range + len as usize))
+
+            // Allocate a Vec<u8> that has at least BUFFER_ALIGNMENT-1 bytes of extra capacity.
+            // This is important as it allows us to return a properly aligned Bytes to the caller,
+            // without needing to write a custom Allocator.
+            let alloc_size =
+                ((len as usize) + (BUFFER_ALIGNMENT - 1)).next_multiple_of(BUFFER_ALIGNMENT);
+            let mut buf = Vec::<u8>::with_capacity(alloc_size as _);
+            let padding = buf.as_ptr().align_offset(BUFFER_ALIGNMENT);
+            unsafe { buf.set_len(padding) };
+
+            let get_range = start_range..(start_range + len as usize);
+
+            let response = object_store
+                .get_opts(
+                    &location,
+                    GetOptions {
+                        range: Some(GetRange::Bounded(get_range)),
+                        ..Default::default()
+                    },
+                )
                 .await?;
-            Ok(bytes)
+
+            let mut byte_stream = response.into_stream();
+            while let Some(bytes) = byte_stream.next().await {
+                let bytes = bytes?;
+                buf.extend_from_slice(&bytes);
+            }
+
+            // bytes_unaligned will contain the entire allocation, so that on Drop the entire buf
+            // is freed.
+            //
+            // bytes_unaligned is a sliced view on top of bytes_unaligned.
```

Comment:
nit: should say "`bytes_aligned` is a sliced view"

Reply:
already fixed ;)